### PR TITLE
HW3

### PR DESCRIPTION
**Pull request by eshcherbin**

### CODE REVIEW

File: speaker_id/speaker_classification/lstm_classifier.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        self.hidden_dim = hidden_dim
+        self.output_dim = output_dim
+
+        self.lstm = nn.LSTM(input_dim, hidden_dim)
+        self.fc = nn.Linear(hidden_dim, output_dim)
+        self.hidden = None
+
+    def init_hidden(self, batch_len, cuda=False):
+        if cuda:
+            self.hidden = (torch.zeros(1, batch_len, self.hidden_dim).cuda(),
```
Comment (Rauf-Kurbanov) [inline speaker_id/speaker_classification/lstm_classifier.py:24]: prefer `.to(device)` over `cuda()`

File: speaker_id/speaker_classification/lstm_classifier.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            self.hidden = (torch.zeros(1, batch_len, self.hidden_dim),
+                           torch.zeros(1, batch_len, self.hidden_dim))
+
+    def forward(self, X, lens):
+        X = pack_padded_sequence(X, lens, batch_first=True)
+        _, (hout, _) = self.lstm(X, self.hidden)
+        out = self.fc(hout[-1])
+        return out
+
+    # def predict(self, X, cuda=False):
```
Comment (Rauf-Kurbanov) [inline speaker_id/speaker_classification/lstm_classifier.py:36]: forgotten code

**Review by Rauf-Kurbanov — APPROVED** [review]
* Do not use absolute paths
* Plots like titles and axis names

Overall nice idiomatic python code: 4/4
